src/main/java/com/proofpoint/discovery/store/HttpRemoteStore.java

```diff
@@ -197,6 +207,20 @@ private static String getHostPort(ServiceDescriptor descriptor)
     @Override
     public void put(Entry entry)
     {
+        if (generalPoolLegacyReplicationMode == ReplicationMode.PHASE_ONE) {
```

Comment:
nit: It would be to move this newly introduced transformation into a separate method.

Reply:
Since phase one (and thus its code) won't last long, I'm not too worried about making its code extra pretty.